packages/core/react/src/WalletProvider.tsx

```diff
@@ -230,8 +238,8 @@ export const WalletProvider: FC<WalletProviderProps> = ({
     }, [isDisconnecting, adapter, setName]);
 
     // Send a transaction using the provided connection
-    const sendTransaction = useCallback(
-        async (transaction: Transaction, connection: Connection, options?: SendTransactionOptions) => {
+    const sendTransaction: WalletAdapterProps['sendTransaction'] = useCallback(
+        async (transaction, connection, options) => {
```

Comment:
The types of the methods in this React context are also not ideal. We don't know if the adapter has `supportedTransactionVersions` or not, so the methods all accept `Transaction | VersionedTransaction` even if this won't work.

Reply:
I think that's ok because the app will be doing runtime checks rather than compile time checks for versioned transaction support for a particular wallet.

Reply:
Oh, I see that this applies to `signTransaction` as well. That's less ideal

Reply:
Yeah. Worse, I think as currently written, it causes `signTransaction` to return `Transaction | VersionedTransaction` which will break the types of all callers.

Reply:
This is fixed now.